### ISSUE TITLE
Make connection handshake timeout configurable

### ISSUE THREAD

**Issue by jhneyra** [rabbitmq/rabbitmq-java-client#81]
Currently I'm trying to connect to RabbitMQ cluster behind HAProxy (also tried with Ngninx 1.9 over TCP) from a Java Client and it takes around 7 seconds to establish the connection.

I noticed that in com.rabbitmq.client.impl.AMQConnection ``connStartBlocker.getReply(HANDSHAKE_TIMEOUT/2)`` is hard-coded to 5 seconds. So, I'm getting Timeout Exception.

I was seeing the code and ``HANDSHAKE_TIMEOUT`` is configured to 10 seconds.
```
public static final int HANDSHAKE_TIMEOUT = 10000;
```

And in the code below ``connStartBlocker.getReply(HANDSHAKE_TIMEOUT/2)`` the reply is waiting for 5 seconds causing the timeout exception.

```
    AMQP.Connection.Start connStart = null;
    AMQP.Connection.Tune connTune = null;
    try {
        connStart =
                (AMQP.Connection.Start) connStartBlocker.getReply(HANDSHAKE_TIMEOUT/2).getMethod();

        _serverProperties = Collections.unmodifiableMap(connStart.getServerProperties());
```

The stack trace exception is
```
java.util.concurrent.TimeoutException
	at com.rabbitmq.utility.BlockingCell.get(BlockingCell.java:77)
	at com.rabbitmq.utility.BlockingCell.uninterruptibleGet(BlockingCell.java:111)
	at com.rabbitmq.utility.BlockingValueOrException.uninterruptibleGetValue(BlockingValueOrException.java:37)
	at com.rabbitmq.client.impl.AMQChannel$BlockingRpcContinuation.getReply(AMQChannel.java:367)
	at com.rabbitmq.client.impl.AMQConnection.start(AMQConnection.java:293)
	at com.rabbitmq.client.ConnectionFactory.newConnection(ConnectionFactory.java:621)
	at com.rabbitmq.client.ConnectionFactory.newConnection(ConnectionFactory.java:648)
	at Consumer.main(Consumer.java:39)
```

I'm using amqp-client 3.5.4 jar. A snippet of the code is:
```
ConnectionFactory factory = new ConnectionFactory();
            factory.setUsername("user");
            factory.setPassword("******");
            factory.setRequestedHeartbeat(58);
            factory.setConnectionTimeout(60000);
            factory.setHost("my.domain.com");
            factory.setPort(5670);
            connection = factory.newConnection();
```

It seems if ``HANDSHAKE_TIMEOUT`` could be configured programmatically in the `` ConnectionFactory`` parameters it should be enough to fix this problem.

**Comment by michaelklishin** [comment]
We can make handshake timeout configurable but 7 seconds to establish a connection suggests something is wrong (or relies on defaults) in the [networking configuration](http://rabbitmq.com/networking.html).

**Comment by jhneyra** [comment]
I tried also with C# client (it takes also around 7-8 seconds to establish connection), however it is not returning TimeoutException like Java client (maybe the HANDSHAKE_TIMEOUT is over 10 seconds, I didn't see the C# code). 

My RabbitMQ client is in Peru and the RabbitMQ Server in US.

**Comment by jhneyra** [comment]
I tried also with C# client (it takes also around 7-8 seconds to establish connection), however it is not returning TimeoutException like Java client (maybe the HANDSHAKE_TIMEOUT is over 10 seconds, I didn't see the C# code). 

My RabbitMQ client is in Peru and the RabbitMQ Server in US.

**Comment by jhneyra** [comment]
I tried also with C# client (it takes also around 7-8 seconds to establish connection), however it is not returning TimeoutException like Java client (maybe the HANDSHAKE_TIMEOUT is over 10 seconds, I didn't see the C# code). 

My RabbitMQ client is in Peru and the RabbitMQ Server in US.

**Comment by michaelklishin** [comment]
Fixed in #82, will be `3.5.5`. Note that the server also has a handshake timeout (configurable) which is 10 seconds by default.
